docs/PortingGuide2-3.md

```diff
@@ -0,0 +1,187 @@
+OpenEXR/Imath 2.x to 3.x Porting Guide
+======================================
+
+This porting guide outlines the several areas where switching from OpenEXR
+2.x to OpenEXR 3.x + Imath 3.x will require source code or build changes of
+downstream software.
+
+In each case, we will often explain both how to change if you are expecting
+3.x only hereafter, or usually a more complex accommodation if you want to
+keep compatibility with both 2.x and 3.x.
+
+
+OpenEXR and Imath are different packages now
+--------------------------------------------
+
+If your use of OpenEXR was only for the sake of using the math classes and
+utilities, maybe you were unhappy that you needed to download and build the
+full OpenEXR dependency. You are in luck -- now Imath is a separate, very
+lightweight open source package. You can use Imath functionality without
+needing any of OpenEXR, which as of 3.x only includes the parts you need to
+read and write OpenEXR image files.
+
+The parts of "IlmBase" that were Imath and `half` are now repackaged as the
+`Imath` library. The IlmThread and Iex libraries have been folded into the
+OpenEXR package, since they were were not necessary to the rest of Imath.
+
+When building OpenEXR 3.x, note that if Imath 3.x library is not found
+already installed at build time, it will be automatically downloaded and
+built as part of the OpenEXR build.
+
+
+
+Finding and using OpenEXR and Imath CMake configs
+-------------------------------------------------
+
+**OpenEXR/Imath 3.x only**
+
+If you are *only* concerned with OpenEXR/Imath 3.x going forward, this is
+the recommended way to find the libraries in a downstream project that uses
+the CMake build system:
+
+    find_package(Imath CONFIG)
```

Comment:
For CONFIG to work, does it imply that the cmake configs got installed for Imath? Should we advise people who are packaging Imath to also package the cmake configs?

Reply:
It would be very rude not to package the cmake configs. Imported configs are the preferred way for modern cmake-based projects to consume their cmake-based dependencies. It's barbaric to expect downstream projects to each write their own buggy FindImath.cmake module.

Reply:
The CONFIG designation is not required... but semantically what it means is "use our installed as the ground truth, don't accidentally use a FindImath.cmake or FindOpenEXR.cmake that some rando on the internet wrote years ago."